lib/puma/reactor.rb

```diff
@@ -18,9 +18,9 @@ class Reactor
     # Create a new Reactor to monitor IO objects added by #add.
     # The provided block will be invoked when an IO has data available to read,
     # its timeout elapses, or when the Reactor shuts down.
-    def initialize(&block)
+    def initialize(backend, &block)
       require 'nio'
-      @selector = NIO::Selector.new
+      @selector = backend == :auto ? NIO::Selector.new : NIO::Selector.new(backend)
```

Comment:
It appears the default option is `:ruby`, not `:auto`. https://github.com/socketry/nio4r/blob/4111b1cee87ce8857558900b8df76a5bd6c3e985/lib/nio/selector.rb#L16

Reply:
Curious on @jcmfernandes 's thoughts here, too, but I kinda like having an explicit default of `:auto`, which means to use whatever `NIO`'s default backend is.

Otherwise, if NIO ever changes their default backend, we end up in a weird situation where puma's default is always `:ruby`, regardless of NIO's actual default.

Reply:
At present, `epoll` is the default backend on Ubuntu.  Windows uses `ruby`.

Reply:
@nateberkopec @cjlarose as stated in https://github.com/socketry/nio4r/pull/258 passing `nil` as a way to say "please choose the best" only works when using libev (well, my PR changes that but it's not even part of a release at this point :smile_cat:).

Hence, either we pass any value set in the puma configuration as is to `NIO::Selector.new` (except for `nil`, which we don't want to pass given I wrote above), or we introduce a value that serves as the "choose the best please". It can be `nil`, but I thought about having something more explicit, hence the `:auto`. Happy to change it though!

Reply:
`:auto` seems fine by me.

Another option to consider is instead of making it a symbol, we just let the user specify a block that would return the instance the user actually wants:

```ruby
# config.rb
nio_selector { NIO::Selector.new(:ruby) }
# or
nio_selector { NIO::Selector.new }
```

This gives the user the ultimately flexibility, without imposing our own defaults or conventions. Using blocks in the config file is already well-established by `on_worker_boot`, `on_worker_shutdown`, etc.

Reply:
@nateberkopec Do you have a preference for using `:auto` to mean "pick the best IO selector" vs using the block for configuration? Or many a different idea?

Reply:
In my head, whatever I provide to `nio_selector` should be passed as an arg to Nio::Selector.new(). So, if you specify `nio_selector :auto`, I expect that to be passed to `NIO::Selector.new`. 

IMO I think the easiest to understand interface here is that the default is nil rather than this new `:auto` symbol, but due to https://github.com/socketry/nio4r/pull/258 I think the best backwards/forwards compatibile option is thew new :auto symbol.

So I'm OK with it.